src/sentry/api/serializers/models/project.py

```diff
@@ -57,55 +59,113 @@
 LATEST_DEPLOYS_KEY = "latestDeploys"
 
 
+def get_access_by_project(
+    projects: Sequence[Project], user: User
+) -> MutableMapping[Project, MutableMapping[str, Any]]:
```

Comment:
This return type is used quite a bit. It might be worth having a common type alias somewhere for it.